src/main/java/org/terasology/launcher/game/Installation.java

```diff
@@ -70,35 +87,50 @@ public String toString() {
                 .toString();
     }
 
+    static Path findJar(Path searchPath, BiPredicate<Path, BasicFileAttributes> predicate, String displayName) throws IOException {
+        Path resultPath;
+
+        try (var jarMatches = Files.find(searchPath, 3, predicate)) {
```

Comment:
If I recall correctly, that `3` is the maximum search depth, and I made it only as deep as I needed to. So I think this shouldn't work when the distribution nests another directory in front.

(I haven't tested it yet. It's certainly possible I miscounted before and the limit is high enough.)

Reply:
:thinking: now I'm not sure anymore - according to _my counting_:tm: level 3 was maximum (and it was previously one too deep), but then I could obviously off-by-one, and we'd need to increase it to 4 here - which brings up the question whether this should be an argument to `findJar`...

Reply:
I managed to run this and it succeeded in finding Terasology.jar. 👍